tests/python/relax/test_runtime_builtin_paged_attention_kv_cache.py

```diff
@@ -0,0 +1,420 @@
+        values = tvm.nd.array(np.expand_dims(global_new_kv[layer_id, 1], axis=0))
+        fappend(cache, f_transpose_append, keys, values, layer_id)
+
+    verify_cached_values(cache, cached_values, f_copy_cache)
+
+    # Pop n
+    for pop_length in [3, 13]:
+        for seq_id in range(nseq):
+            fpopn(cache, seq_id, pop_length)
+            cached_values[seq_id] = cached_values[seq_id][:, :, :-pop_length, ...]
+    fsync(cache)
+    verify_cached_values(cache, cached_values, f_copy_cache)
+
+    # Decode after pop n
+    for _ in range(5):
+        decode_new_kv = np.random.rand(nlayer, 2, nseq, 1, nhead, nfeat).astype(dtype)
+        freset_append_length(cache)
+        for seq_id in range(nseq):
+            freserve(cache, seq_id, 1)
+        fsync(cache)
+
+        for seq_id in range(nseq):
+            cached_values[seq_id] = np.concatenate(
+                [cached_values[seq_id], decode_new_kv[:, :, seq_id, ...]], axis=2
+            )
+        for layer_id in range(nlayer):
+            keys = tvm.nd.array(decode_new_kv[layer_id, 0])
+            values = tvm.nd.array(decode_new_kv[layer_id, 1])
+            fappend(cache, f_transpose_append, keys, values, layer_id)
+
+        verify_cached_values(cache, cached_values, f_copy_cache)
+
+
+def test_paged_attention_kv_cache_clear():
+    f_transpose_append, f_copy_cache = build_tir_func([transpose_append, copy_cache])
+    cache = fcreate(
+        tvm.runtime.ShapeTuple([reserved_nseq, total_seq_len, page_size]),
+        nlayer,
+        nhead,
+        nfeat,
+        tvm.nd.empty((), dtype),
+    )
+
+    cached_values = []
+    initial_lengths = [20, 24, 26, 27]
+
+    # Initial prefill
+    freset_append_length(cache)
+    for seq_id, append_length in enumerate(initial_lengths):
+        seq_id_in_cache = fadd_sequence(cache)
+        assert seq_id_in_cache == seq_id
+        freserve(cache, seq_id, append_length)
+    fsync(cache)
+
+    global_new_kv = np.zeros((nlayer, 2, 0, nhead, nfeat), dtype)
+    for length in initial_lengths:
+        new_kv = np.random.rand(nlayer, 2, length, nhead, nfeat).astype(dtype)
+        cached_values.append(new_kv)
+        global_new_kv = np.concatenate([global_new_kv, new_kv], axis=2)
+    for layer_id in range(nlayer):
+        keys = tvm.nd.array(np.expand_dims(global_new_kv[layer_id, 0], axis=0))
+        values = tvm.nd.array(np.expand_dims(global_new_kv[layer_id, 1], axis=0))
+        fappend(cache, f_transpose_append, keys, values, layer_id)
+
+    verify_cached_values(cache, cached_values, f_copy_cache)
+
+    # Clear
+    fclear(cache)
+    verify_cached_values(cache, [], f_copy_cache)
+
+
+if __name__ == "__main__":
+    test_paged_attention_kv_cache_append_prefill()
+    test_paged_attention_kv_cache_append_decode()
+    test_paged_attention_kv_cache_remove()
+    test_paged_attention_kv_cache_popn()
+    test_paged_attention_kv_cache_clear()
+    # tvm.testing.main()
```

Comment:
how about uncomment this test.

BTW, looks the test case for "vm.builtin.paged_attention_kv_cache_attention" is missing

Reply:
We have the attention part tested with manual CUDA code developed by @yzh119 (great thanks!). We don't have TIR attention functions yet so it cannot be tested in a tidy way right now in this file. As a result, the attention tests are skipped for now. I just added a comment on that. We will follow up with the tests once we have the attention written in TIR.